src/components/Todos/AddTodoForm.js

```diff
@@ -0,0 +1,79 @@
+import Card from "../UI/Card";
+import { useState, useRef, useContext } from "react";
+import { useHistory } from "react-router";
+import { TodoContext } from "../../store/todo-context";
+import { AuthContext } from "../../store/auth-context";
+
+const validateData = (task, dueDate) => {
```

Comment:
Validaci bych spíš řešil v `todoCtx.addTodo`. Takhle bys to musel kopírovat, když bys `addTodo` používal na více místech.